### PR TITLE
Proxy everything

### LINKED ISSUE #2643: Updates should be proxyfied
If we have a proxy set, put all our QNetworkAccessManagerS under proxy, including updates, toxme, etc.

### PR DESCRIPTION
**Pull request by sudden6**
Use the set proxy also for autoupdater and toxme lookups.

fixes #2643

### CODE REVIEW

File: src/net/autoupdate.cpp [hunk 1]
```diff
@@ -108,6 +108,7 @@ AutoUpdater::VersionInfo AutoUpdater::getUpdateVersion()
 
     QNetworkAccessManager *manager = new QNetworkAccessManager;
     QNetworkReply* reply = manager->get(QNetworkRequest(QUrl(checkURI)));
+    manager->setProxy(Settings::getInstance().getProxy());
```
Comment (tux3) [inline src/net/autoupdate.cpp:110]: No! 
manager->get() will immediately start the request, at best you're racing to set the proxy, but it'll probably just not be proxied at all.

**Comment by sudden6** [comment]
@tux3 fixed thanks

**Comment by zetok** [comment]
@sudden6 btw, to avoid commits like 98f49c4494bb3c9a88ea2bb2fabde21d3e463e16, you can just simply rebase your branch - e.g. in this case it would be `git rebase master proxy-everything`

**Comment by sudden6** [comment]
@zetok Thanks, does this also work if I apply it after a merge commit appears? e.g if I now do `git rebase master proxy-everything` the merge commit will disapear?

**Comment by tux3** [comment]
No, the merge commit won't disappear now. You could attempt to remove the commit and then rebase again, but that's nontrivial.

**Comment by sudden6** [comment]
ok, so this PR is ok now?

**Comment by tux3** [comment]
LGTM

**Pull request merged**